WalletConnectSharp.Core/Models/MessageHandler/TypedEventHandler.cs

```diff
@@ -258,5 +258,11 @@ async Task<VerifiedContext> VerifyContext(string hash, Metadata metadata)
 
             return context;
         }
+
+        public virtual void Dispose()
+        {
+            var context = _ref.Context;
+            _instances.Remove(context);
+        }
```

Comment:
Please implement disposal with `Dispose(bool)` overload

https://learn.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose#dispose-and-disposebool